src/ripple/protocol/impl/STAmount.cpp

```diff
@@ -1250,22 +1267,12 @@ divRound (STAmount const& num, STAmount const& den,
         }
 
     bool resultNegative = num.negative() != den.negative();
-    // Compute (numerator * 10^17) / denominator
-    CBigNum v;
-
-    if ((BN_add_word64 (&v, numVal) != 1) || (BN_mul_word64 (&v, tenTo17) != 1))
-        Throw<std::runtime_error> ("internal bn error");
-
-    if (resultNegative != roundUp) // Rounding down is automatic when we divide
-        BN_add_word64 (&v, denVal - 1);
 
-    if (BN_div_word64 (&v, denVal) != 1)
-        Throw<std::runtime_error> ("internal bn error");
-
-    // 10^16 <= quotient <= 10^18
-    assert (BN_num_bytes (&v) <= 64);
+    // Compute (numerator * 10^17) / denominator
+    std::uint64_t amount = fast_muldiv (
+        numVal, tenTo17, denVal,
+        (resultNegative != roundUp) ? denVal - 1 : 0);
```

Comment:
This too looks like round-away-from-zero to me.  So the same notes as on line 1215 apply.